### DISCUSSION TITLE
How to use SCPI to communicate with LibreVNA without opening LibreVNA-GUI

### DISCUSSION THREAD

**Discussion by pishuai1995** [jankae/LibreVNA#218] [Q&A]
Hello,
I am a little confused, if I write my own script to control LibreVNA, do I have to open LibreVNA-GUI?  Currently, I cannot connect to LibreVNA when the Librevna-GUI is not open.

**Comment by jankae** [comment]
Hi, that is normal. The SCPI server is run by the LibreVNA-GUI. You always have to open the GUI, but you can start it with the "--no-gui" argument which prevents any windows from showing up (but the code for the windows is still running in the background)

**Reply by pishuai1995** [reply]
Thank you for your prompt reply. Sorry to bother you. I would like to ask, how can I start it with the "--no-gui" argument？

**Reply by pishuai1995** [reply]
Is this a cmd directive?

**Reply by jankae** [reply]
Yes, you need to add the argument to the program call on the command line: `LibreVNA-GUI --no-gui` should do the trick.

**Reply by pishuai1995** [reply]
Thank you ! I understand.

**Comment by ozzyys** [comment]
Hello if i start this command "LibreVNA-GUI --no-gui" from my console. It works no problem.

But if i start this command like this from python script, os.system("LibreVNA-GUI --no-gui") the program don't start. So python script also don't make nothing.

**Reply by jankae** [reply]
I am sure your python script will spit out some kind of error message if it can't start the GUI. This error message will be very helpful here.

In any case, this is my very minimal example to test this:
First, starting the GUI with visible windows:
```python
import os
os.system("./LibreVNA-GUI")
```
And now with the `--no-gui` option:
```python
import os
os.system("./LibreVNA-GUI --no-gui")
```

Both work as intended. The python script is located (and executed) in the same folder as the LibreVNA-GUI executable. Otherwise you might see something like this:
```bash
sh: 1: ./LibreVNA-GUI: not found
```
